File: src/main/java/fds/java/consistent/RingNode.java
package fds.java.consistent;

/**
 * Most simple local representation of a node on teh Ring
 */
import java.util.HashMap;
import java.util.Map;


public class RingNode {

    private String name;

    private Map<String,Object> map = new HashMap<String,Object>();

    public RingNode(String name) {
        setName(name);
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public void put(String  key, Object  value) {
        map.put(key,value);
    }

    public Object get( String key) {
        return map.get(key);
    }

    public int size() {
        return map.size();
    }

}

File: src/main/scala/fds/scala/consistent/RingNode.scala
package fds.scala.consistent

import scala.collection.mutable.Map

/**
  * Most simple local representation of a node on the Ring
  */
class RingNode(val name: String) {
  private val map: Map[String, Any] = Map[String, Any]()

  def put(key: String, value: Any): Unit = {
    map.put(key, value)
  }

  def get(key: String): Any = map.get(key)

  def size: Int = map.size
}

object RingNode {
  def apply(name: String): RingNode = new RingNode(name)
}